### PR TITLE
Added string tokenizer in java

### PR DESCRIPTION
**Pull request by toihirhalim**
# Java String Tokenizer
#### changes:
* new file tokenizer.java
* link the file in readme.md

#### how to call the function
tokenize(str, delimiters)
* **str** : is a text
* **delimiters** : is a list of char which is by default white space  ' '

exemple: 
```
tokenize("hello world") = tokenize("hello world", ' ') = [ 'hello', 'world' ]
tokenize("hello world", ' ', 'l') = [ 'he', 'o', 'wor', 'd' ]
```

### CODE REVIEW

File: strings/README.md [hunk 1]
```diff
@@ -24,6 +24,7 @@
 2. [All subsequences](java/sequence.java)
 3. [KMP String Searching](java/kmp.cpp)
 4. [Rabin Karp String Searching](java/rabin-karp.cpp)
+4. [String Tokenizer](java/tokenizer.java)
```
Comment (TawfikYasser) [inline strings/README.md:27]: Hi @toihirhalim Please convert `4` to `5` because there is already a program with number 4

Comment (toihirhalim) [reply strings/README.md:27]: Hi @TawfikYasser  
I'm changing it now

**Review by TawfikYasser — APPROVED** [review]
Great work @toihirhalim

**Comment by toihirhalim** [comment]
thank you @TawfikYasser

**Pull request merged**